tests/queries/0_stateless/02403_enable_extended_results_for_datetime_functions.sql.j2

```diff
@@ -5,5 +5,10 @@ SELECT 'type;{{ date_fun }};toDate32;{{ option_value }}', toTypeName({{ date_fun
 SELECT '{{ date_fun }};toDateTime64;{{ option_value }}', {{ date_fun }}(toDateTime64('1920-02-02 10:20:30', 3)) SETTINGS enable_extended_results_for_datetime_functions = {{ option_value }};
 SELECT 'type;{{ date_fun }};toDateTime64;{{ option_value }}', toTypeName({{ date_fun }}(toDateTime64('1920-02-02 10:20:30', 3))) SETTINGS enable_extended_results_for_datetime_functions = {{ option_value }};
 {% endfor -%}
+{% for date_fun in ['toStartOfDay', 'toStartOfHour', 'toStartOfMinute', 'toStartOfFiveMinutes', 'toStartOfTenMinutes', 'toStartOfFifteenMinutes', 'timeSlot'] -%}
```

Comment:
I am not sure if such tests exist already, but it would be cool to have a few tests for "boundary cases", i.e. the absolute min/max values for each input value range. If it is inconvenient to integrate them into this test, a separate SQL test is also good.

Reply:
I am also not sure if there are tests for  "boundary cases" but after code review I found several boundary issues in about 5-6 date/time functions. I am going to create a mini PR to fix these boundary issues. So PR will surely have the boundary test.